app/javascript/containers/FloofIndexContainer.js

```diff
@@ -5,28 +5,59 @@ class FloofIndexContainer extends Component {
   constructor(props){
     super(props)
     this.state = {
-      floofs: [{
-        id: 1,
-        name: 'Spot',
-        job_title: 'Senior Web Developer',
-        picture: 'http://mymagicdog.com/wp-content/uploads/2016/06/dog-typing-2.jpg'
-      },
-      {
-        id: 2,
-        name: 'Rova',
-        job_title: 'Junior Associate',
-        picture: 'http://teachyourpetstotype.com/dogtyping4.jpg'
-      }
-    ]
+      floofs: []
     }
   }
 
+  componentDidMount() {
+    fetch('/api/v1/floofs')
+    .then(response => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+            error = new Error(errorMessage);
+          throw(error);
+        }
+      })
+      .then(response => {
+        return response.json()
+      })
+      .then(body => {
+        this.setState({ floofs: body })
+      })
+      .catch(error => console.error(`Error in fetch: ${error.message}`));
+  }
+
   render() {
+    let floofData = this.state.floofs
+    let floofCatTileComponents;
+
+    if (floofData !== []) {
+      //create a list of categories
+      let categories = floofData.map((floof) => {
+        return floof.category
+      })
+      categories = categories.filter((category, index, categories) => categories.indexOf(category) === index);
+      categories = categories.sort()
+      
+      //map through our list of categories
```

Comment:
this processing and work should be done on the API side - you can leverage the power of the database to help with this.